xbmc/video/tags/VideoInfoTagLoaderFactory.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+ *      Copyright (C) 2005-2013 Team XBMC
+ *      http://xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "VideoInfoTagLoaderFactory.h"
+#include "VideoTagLoaderFFmpeg.h"
+#include "VideoTagLoaderNFO.h"
+#include "addons/AudioDecoder.h"
+
+using namespace VIDEO;
+
+IVideoInfoTagLoader*
+CVideoInfoTagLoaderFactory::CreateLoader(const CFileItem& item,
+                                         ADDON::ScraperPtr info,
+                                         bool lookInFolder)
+{
+  // don't try to read tags for streams
+  if (item.IsInternetStream())
+    return nullptr;
+
+  if (item.IsType(".mkv") || item.IsType(".mp4"))
+  {
+    CVideoTagLoaderFFmpeg* ff = new CVideoTagLoaderFFmpeg(item, info, lookInFolder);
+    if (ff->HasInfo());
```

Comment:
semicolon at the end of the line looks wrong

Reply:
indeed.